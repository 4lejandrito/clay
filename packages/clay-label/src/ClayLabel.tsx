/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import classNames from 'classnames';
import Icon from '@clayui/icon';

type DisplayType = 'secondary' | 'info' | 'warning' | 'danger' | 'success';

interface Props
	extends React.BaseHTMLAttributes<HTMLAnchorElement | HTMLSpanElement> {
	/**
	 * HTML properties that are applied to the 'x' button.
	 */
	closeButtonProps?: React.ButtonHTMLAttributes<HTMLButtonElement>;

	/**
	 * Determines the style of the label.
	 */
	displayType?: DisplayType;

	/**
	 * Flag to indicate if the label should be of the `large` variant.
	 */
	large?: boolean;
}

const ClayLabel: React.FunctionComponent<Props> = ({
	children,
	className,
	closeButtonProps,
	displayType = 'secondary',
	href,
	large = false,
	...otherProps
}) => {
	const TagName = href ? 'a' : 'span';

	return (
		<TagName
			{...otherProps}
			className={classNames('label', className, {
				'label-dismissible': closeButtonProps,
				'label-lg': large,
				[`label-${displayType}`]: displayType,
			})}
			href={href}
		>
			<span className="label-item label-item-expand">{children}</span>

			{closeButtonProps && (
				<span className="label-item label-item-after">
					<button
						{...closeButtonProps}
						className="close"
						type="button"
					>
						<Icon symbol="times" />
					</button>
				</span>
			)}
		</TagName>
	);
};

export default ClayLabel;
