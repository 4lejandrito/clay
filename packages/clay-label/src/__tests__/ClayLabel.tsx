/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ClayLabel from '../ClayLabel';

describe('ClayLabel', () => {
	it('renders', () => {
		const testRenderer = TestRenderer.create(
			<ClayLabel>{'Default Label'}</ClayLabel>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders with a different displayType ', () => {
		const testRenderer = TestRenderer.create(
			<ClayLabel displayType="success">{'Success Label'}</ClayLabel>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as a link ', () => {
		const testRenderer = TestRenderer.create(
			<ClayLabel href="#/foo/bar">{'Label w/ link'}</ClayLabel>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});
