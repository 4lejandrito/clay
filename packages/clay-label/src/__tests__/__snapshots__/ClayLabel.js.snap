// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayLabel should render a \`danger color\` label 1`] = `
<span class="label label-danger">
  <span class="text-truncate-inline">
    <span class="text-truncate"></span>
  </span>
</span>
`;

exports[`ClayLabel should render a \`default color\` label 1`] = `
<span class="label label-secondary">
  <span class="text-truncate-inline">
    <span class="text-truncate"></span>
  </span>
</span>
`;

exports[`ClayLabel should render a \`info color\` label 1`] = `
<span class="label label-info">
  <span class="text-truncate-inline">
    <span class="text-truncate"></span>
  </span>
</span>
`;

exports[`ClayLabel should render a \`success color\` label 1`] = `
<span class="label label-success">
  <span class="text-truncate-inline">
    <span class="text-truncate"></span>
  </span>
</span>
`;

exports[`ClayLabel should render a \`warning color\` label 1`] = `
<span class="label label-warning">
  <span class="text-truncate-inline">
    <span class="text-truncate"></span>
  </span>
</span>
`;

exports[`ClayLabel should render a label with classes 1`] = `
<span class="label label-secondary my-custom-class">
  <span class="text-truncate-inline">
    <span class="text-truncate"></span>
  </span>
</span>
`;

exports[`ClayLabel should render a label with id 1`] = `
<span class="label label-secondary" id="myId">
  <span class="text-truncate-inline">
    <span class="text-truncate"></span>
  </span>
</span>
`;

exports[`ClayLabel should render a label with label 1`] = `
<span class="label label-secondary">
  <span class="text-truncate-inline">
    <span class="text-truncate">Foo</span>
  </span>
</span>
`;

exports[`ClayLabel should render a label with link 1`] = `
<span class="label label-secondary">
  <span class="text-truncate-inline">
    <span class="text-truncate">
      <a href="#1">Foo</a>
    </span>
  </span>
</span>
`;

exports[`ClayLabel should render a label with removable 1`] = `
<span class="label label-secondary label-dismissible">
  <span class="label-item label-item-expand">
    <span class="text-truncate-inline">
      <span class="text-truncate">Foo</span>
    </span>
  </span>
  <span class="label-item label-item-after">
    <button class="btn close btn-unstyled" aria-label="Close" type="button">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times" focusable="false">
        <title>times</title>
        <use xlink:href="icons.svg#times"></use>
      </svg>
    </button>
  </span>
</span>
`;

exports[`ClayLabel should render a large label 1`] = `
<span class="label label-secondary label-lg">
  <span class="text-truncate-inline">
    <span class="text-truncate"></span>
  </span>
</span>
`;

exports[`ClayLabel should render the default markup 1`] = `
<span class="label label-secondary">
  <span class="text-truncate-inline">
    <span class="text-truncate"></span>
  </span>
</span>
`;
